deepsubtree.go

```diff
@@ -33,7 +33,7 @@ func (dsmst *DeepSparseMerkleSubTree) AddBranch(proof SparseMerkleProof, key []b
 	}
 
 	if valueHash != nil {
-		if err := dsmst.values.Set(valueKey(key, valueHash), value); err != nil {
+		if err := dsmst.values.Set(dsmst.SparseMerkleTree.th.path(key), value); err != nil {
```

Comment:
Correct me if I'm wrong, but embedding is done explicitly to avoid having to use `.SparseMerkleTree`. It seems more idiomatic to actually use embedding as intended here.

```suggestion
		if err := dsmst.values.Set(dsmst.th.path(key), value); err != nil {
```